katsdpcontroller/generator.py

```diff
@@ -1578,14 +1579,22 @@ async def _make_spectral_imager(g: networkx.MultiDiGraph,
     dump_bytes = stream.vis.n_baselines * defaults.SPECTRAL_OBJECT_CHANNELS * BYTES_PER_VFW_SPECTRAL
     data_url = _stream_url(capture_block_id, stream.name + '.' + stream.vis.name)
     targets, data_set = _get_targets(configuration, capture_block_id, stream, telstate_endpoint)
-    band = data_set.spectral_windows[data_set.spw].band
+    spw = data_set.spectral_windows[data_set.spw]
+    band = spw.band
     channel_freqs = data_set.channel_freqs * u.Hz
     del data_set    # Allow Python to recover the memory
 
     async with katsdpmodels.fetch.aiohttp.TelescopeStateFetcher(telstate) as fetcher:
         rfi_mask = await fetcher.get('model_rfi_mask_fixed', RFIMask)
         max_baseline = rfi_mask.max_baseline_length(channel_freqs)
         channel_mask = max_baseline > 0 * u.m
+        acv = stream.vis.baseline_correlation_products.antenna_channelised_voltage
+        telstate_acv = telstate.view(acv.name)
+        band_mask = await fetcher.get('model_band_mask_fixed', BandMask, telstate=telstate_acv)
+        channel_mask |= band_mask.is_masked(
+            SpectralWindow(spw.bandwidth * u.Hz, spw.centre_freq * u.Hz),
```

Comment:
I've realised that this is wrong - the bandwidth of the data that reaches katdal might not be the full band. I'll make an update that instead gets the data from the CAM sensors.